packages/material-ui/src/styles/createTypography.d.ts

```diff
@@ -1,7 +1,7 @@
 import { Palette } from './createPalette';
 import { CSSProperties } from './withStyles';
 
-export type ThemeStyle =
+export type Variant =
```

Comment:
Would this not cause issues for people using this type in a TypeScript environments?  Or should consumers not be using these types directly?

Reply:
As far as I know you don't have access to this type from /core or /core/styles so this should be safe. Since this is a type and not an interface it's also not subject to module augmentation.

Reply:
So doing something like this:
```
import { ThemeStyle, TypographyStyle } from '@material-ui/core/styles/createTypography';
```
Would not be recommended?

Reply:
Any module that is more than 1 level deep is private, for instance `createTypography`.

Reply:
@oliviertassinari Good to know. I actually was using this type to type a variant in a wrapper i made which took a variant, so this did break that. Doesn't look like there is any base exported text variant type. What would you recommend i do here? Just redefine `Variant `as my own type?

Reply:
I have seen that we document nested imports in the TypeScript section, so maybe we enforce a different rule for TypeScript and it's fine.
Alternatively, I imagine we could re-export the types that developers might need.

Reply:
Either one would be super helpful. Thanks for considering!q

Reply:
added a feature request https://github.com/mui-org/material-ui/issues/19572